### PR TITLE
Feat/final-deliverable

### PR DESCRIPTION
**Pull request by enriquedfa**
Added test file to check functuonality of homepage, product page, product detail page, search page and the shopping cart.

Not sure if it's my internet but it works 8 out of 10 executions since sometimes stuff from the API do not load on time. :( 

Note: Although the tests are being executed correctly the terminal has a bunch of errors coming from the API fetching hooks, currently working on that.
Also, I saw the tip about using msw very late, I did try to use it but with time low I wasn't successful on that.

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -1,17 +1,312 @@
...
+      expect(cart).toBeInTheDocument();
+      expect(cart).toHaveTextContent("Cart (48)");
+
+      expect(addToCart).toBeDisabled();
+    });
+  });
+});
+
+// TODO - It doesn't work, I don't know why
+xdescribe("Search Results Page", () => {
```
Comment (YeipiLozano) [inline src/App.test.js]: it says xdescribe

Comment (enriquedfa) [reply src/App.test.js]: Check the latest commit

Comment (YeipiLozano) [reply src/App.test.js]: ay sisierto q burro soy XD

File: src/components/CartItem/CartItem.jsx [hunk 2]
```diff
@@ -61,7 +69,7 @@ function CartItem({ item, stock, onRemoveItem, onChangeQuantity }) {
         </div>
       </div>
       <div className="cart-item-total">
-        <div className="cart-item-total-value">
+        <div className="cart-item-total-value" title="Item Subtotal">
           Total: $ {item.price * item.quantity}
```
Comment (danbenitezf) [inline src/components/CartItem/CartItem.jsx]: I may forgot to add this in the previous review but, better store this total on a variable

File: src/App.test.js [hunk 3]
```diff
@@ -1,17 +1,297 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import {
```
Comment (danbenitezf) [inline src/App.test.js]: well first suggestion is don't create the test for all components on the same file. It is way easier to find, update and manage the tests if you divide it into it's corresponding file. What is usually done if you have a `RandomComponent.jsx` create a test file just for it, something like `RandomComponent.test.js` 
Also there are multiple approaches on where to put this files one is to have the test directly on the directory the component is and the other one is in the `src` create a new folder `tests` then you basically copy your `src` structure, if in src you have `src/components` or `src/hooks` on the tests folder you would create a `src/tests/components`

File: src/components/ShoppingCart/ShoppingCart.js [hunk 4]
```diff
@@ -71,7 +80,7 @@ function ShoppingCart() {
       <div className="shopping-cart__footer">
         <div className="shopping-cart__footer__total">
           <h2>Total</h2>
-          <p>$ {state.total}</p>
+          <p title="Cart Total">$ {state.total}</p>
           <Link to="/checkout">
             <button disabled={state.cart.length === 0}>Checkout</button>
```
Comment (danbenitezf) [inline src/components/ShoppingCart/ShoppingCart.js]: imo its better to have this evaluation on a variable

**Review by danbenitezf — COMMENTED** [review]
I don't think you will be able to add my suggestions for the closing session (mb it took me a lot of time to review it), if you want to apply them later i think we can review them on slack.
Well done all bootcamp.

File: src/components/ShoppingCart/ShoppingCart.js (hunk 4 above)
Comment (enriquedfa) [reply src/components/ShoppingCart/ShoppingCart.js]: So having a constant with the title?

`<p title={title_CartTotal}>$ {state.total}</p>`

Comment (danbenitezf) [reply src/components/ShoppingCart/ShoppingCart.js]: I put the comment on the wrong line, my bad. I was intended to go on 85
```
<button disabled={state.cart.length === 0}>Checkout</button>
```
to be something like 
```
<button disabled={disableCheckout}>Checkout</button>
```

Comment (enriquedfa) [reply src/components/ShoppingCart/ShoppingCart.js]: Oooh got it, you're right, I moved that operation to its own constant

File: src/components/CartItem/CartItem.jsx (hunk 2 above)
Comment (enriquedfa) [reply src/components/CartItem/CartItem.jsx]: Got it!

File: src/App.test.js (hunk 3 above)
Comment (enriquedfa) [reply src/App.test.js]: All right, sounds good, I'll try to do it asap

Comment (enriquedfa) [reply src/App.test.js]: I separated the test to their own component, working good for now

**Pull request merged**